Sources/SwiftProtobuf/Message+JSONAdditions.swift

```diff
@@ -90,23 +93,22 @@ extension Message {
   ) throws {
     self.init()
     try jsonUTF8Data.withUnsafeBytes { (body: UnsafeRawBufferPointer) in
-      if body.count > 0 {
-        var decoder = JSONDecoder(source: body, options: options)
-        if !decoder.scanner.skipOptionalNull() {
-          try decoder.decodeFullObject(message: &self)
-        } else if Self.self is _CustomJSONCodable.Type {
-          if let message = try (Self.self as! _CustomJSONCodable.Type)
-            .decodedFromJSONNull() {
-            self = message as! Self
-          } else {
-            throw JSONDecodingError.illegalNull
-          }
+      guard body.count > 0 else {
+        throw JSONDecodingError.truncated
```

Comment:
I wonder if this should get a comment since zero length makes sense for binary?

Reply:
Done.